roles/bind-keycloak-apb/tasks/main.yml

```diff
@@ -1,89 +1,77 @@
 - name: Generate client id
   shell: tr -d -c "a-zA-Z0-9" < /dev/urandom | head -c 20
   register: generated_client_id
-- set_fact: client_id={{ generated_client_id.stdout }}
 
 - name: Generate client secret
   shell: tr -d -c "a-zA-Z0-9" < /dev/urandom | head -c 20
-  register: generated_secret
+  register: generated_client_secret
 
 - name: "Retrieve route to keycloak"
-  shell: "oc get configmaps/{{keycloak_service_name}} -n {{ namespace }} -o jsonpath='{.data.uri}'"
+  shell: "oc get configmaps/{{keycloak_service_name}} -n {{ namespace }} -o jsonpath={.data.uri}"
   register: keycloak_route
 
+- name: "Retrieve route to keycloak"
+  shell: "oc get configmaps/{{keycloak_service_name}} -n {{ namespace }} -o jsonpath={.items[*].data.realm}"
+  register: keycloak_kcdev_realm
+
+- set_fact:
+    KEYCLOAK_REALM: {{ REALM_NAME is defined | ternary(REALM_NAME, namespace) }}
+    KEYCLOAK_URI: {{ keycloak_route.stdout }}
+    CLIENT_ID: {{ generated_client_id.stdout }}
+    CLIENT_SECRET: {{ generated_client_secret.stdout }}
+
+- name: Check for {{ KEYCLOAK_REALM }} is same as we passed for binding
```

Comment:
I thought we were going to remove the `REALM_NAME` from the bind parameters?

Reply:
Done!